app/Http/Controllers/PostController.php

```diff
@@ -362,22 +362,36 @@ public function flag(Request $request, Post $post)
      */
     public function unflag(Request $request, Post $post)
     {
-        if ($post->flags == 1)
+        if ($post->flags > 0)
         {
-            $post->decrement('flags');
-        $user = Auth::user();
-        $user->flags()->detach($post->id);
+            $post->flags = 0;
+            $post->save();
+            $user = Auth::user();
+            $user->Post_flags()->attach($post->id, ["reason" => $request->get('content') == null ? "this is a placeholder" : $request->get('content'), "action" => 0, "add_flag_id" => $request->get('flag_id')]);
```

Comment:
I'm not really happy with "this is a placeholder", perhaps we could check the `$request->get('content')` for if it's `empty` and if it is we return "no reason given" instead.